<launch>
<!-- Top-Level launch file for bundled launching /-->
<!-- Launcher for the uav_ftc package -->

	<!-- Available arguments -->
	<arg name="manualMode" default="false"/>
	<arg name="autoMode" default="false"/>
	<arg name="uav_name" default="default"/>
	<arg name="log" default="true"/>

    <!-- Launch uav_ftc-related nodes -->
    <group ns="$(arg uav_name)">
        <!-- Load controller parameters -->
        <param name="uav_name" value="$(arg uav_name)"/>
        <rosparam>
            ctrlMode: 1
            referenceMin: [-1.5, -1.5, -0.25]
            referenceMax: [1.5, 1.5, 0.25]
            wp: 10
            wq: 10
            wr: 10
            wda: 100
            wde: 100
            wdr: 10
            wnp: 1000
            wnq: 1000
            wnr: 1000
        </rosparam>

        <!--  -->
        <!-- Launch data sources -->
        <!--  -->
        <include file="$(find uav_ftc)/launch/data_sources.launch" />

        <!-- Raise rate controller -->
        <node pkg="uav_ftc" name="rate_controller" type="rate_controller_pid" respawn="true" output="screen">
            <param name="~rate" value="50"/>
            <param name="~kp_p" value="0.6"/>
            <param name="~ki_p" value="0.0"/>
            <param name="~kd_p" value="0"/>
            <param name="~kp_q" value="0.6"/>
            <param name="~ki_q" value="0.0"/>
            <param name="~kd_q" value="0"/>
            <param name="~kp_r" value="1.2"/>
            <param name="~ki_r" value="0.0"/>
            <param name="~kd_r" value="0"/>
        </node>

        <!-- Raise input aggreggator node -->
        <node pkg="uav_ftc" name="input_aggregator" type="input_aggregator" respawn="true" output="screen">
            <param name="~rate" value="100"/>
        </node>

        <!-- Raise reference generator node -->
        <node pkg="uav_ftc" name="reference_generator" type="reference_generator" respawn="true" output="screen">
            <remap from="~ctrlMode" to="ctrlMode"/>
            <remap from="~referenceMin" to="referenceMin"/>
            <remap from="~referenceMax" to="referenceMax"/>
            <remap from="refCmds" to="refRates"/>
        </node>

        <!-- Raise rosflight topic converted node -->
        <node pkg="uav_ftc" name="rosflight_publisher" type="rosflight_publisher" respawn="true" output="screen"/>

        <!--  -->
        <!-- Launch logger -->
        <!--  -->
        <include if="$(arg log)" file="$(find uav_ftc)/launch/logger.launch" />

    </group>

</launch>